integr8ly/installation
# Intly 3625 generic resource overrides all ns

## Additional Information
https://issues.jboss.org/browse/INTLY-3625 - support resource requests/limits/replicas overrides for scaling purposes, needed to cover test scenarios for CS-SRE testing of the update_resources playbook.
Some refactoring had to be done due to variable naming conflicts in the middleware-monitoring and mobile-related projects which were using the <role_prefix>_resources variable to refer to multiple resource files internally.

There is also quite a bit of hardening and efficiency improvements added which will automatically be exercised as long as one of the default resources blocks are left empty (all but 3scale, enmasse, sso/user-sso).

## Verification Steps
Cluster should be running at least 1.4 or higher.
1. Log into cluster with oc as user with cluster-admin permissions
2. Select a couple of requests and/or limits values to override across any of the roles (see roles/<role_name>/defaults/main.yml for managed products)
3. For each of the app components selected, either:
a. uncommenting the default values for the block and changing one or more of the values 
b. create a external vars file and define some overrides in that file (Examples can be found at: https://gist.github.com/jessesarn/91a72abb00c30acceeaf71862cb20fa6)
4. Run the playbook using one of the following cmds based on which option you used above:
```
# if overrides made within default vars files
ansible-playbook -i inventories/hosts playbooks/update_resources.yml -vvv -e ns_prefix=""

# if using external vars file
ansible-playbook -i inventories/hosts playbooks/update_resources.yml -vvv -e ns_prefix="" -e @<path/to/vars/file.ext>
```
5. Verify that the resources of the deploys/dcs/statefulsets reflect the overrides you specified.

## Is an upgrade task required and are there additional steps needed to test this?
